### ISSUE TITLE
[QUESTION] RoPE calculation

### ISSUE THREAD

**Issue by ZhihaoLi1** [thunlp/Seq1F1B#6]
**Describe the bug**
rope part logic not considered

**To Reproduce**
Steps to reproduce the behavior. The easier it is to reproduce the faster it will get maintainer attention.
pretrain or finetune on LLAMA checkpoint

**Expected behavior**
**when seq1f1b is enabled, the initial training loss should be around 2, while it starts with 11.4 . With seq1f1b being disabled, initial loss gets reasonable.**

**Stack trace/logs**
If applicable, add the stack trace or logs from the time of the error.

**Environment (please complete the following information):**
 - Megatron-LM commit ID
 - PyTorch version
 - CUDA version
 - NCCL version
Megatron 0.12.1
pytorch 2.5.0

**Proposed fix**
If you have a proposal for how to fix the issue state it here or link to a PR.
Should add offset to the RoPE calculation.

**Additional context**
not sure if the solution can also be applicable to MLA/MoE architecture.

**Comment by MayDomine** [comment]
Yes, the RoPE calculation could potentially be an issue with Seq1F1B. Unfortunately, I currently don’t have much time to address this. However, I’d be happy to review and test any PR if you’re willing to submit to resolve this.

**Comment by MayDomine** [comment]
I have already fixed this; it should work fine now. 
Please try again and let me know if the loss is reasonable.
refer to this commit : https://github.com/thunlp/Seq1F1B/commit/489148ccb400f48b594ef706d49cad0909f5e485
